contributors/devel/sig-instrumentation/migration-to-structured-logging.md

```diff
@@ -214,37 +220,85 @@ func LogHTTP(r *http.Request) {
 }
 ```
 
-Names of arguments should use [lowerCamelCase] and be alphanumeric. Arguments names in one log call should be unique.
-Names should be picked based on semantic meaning of value itself, not the context in which is used (log message should
-imply the context). For example names like `status` should be used over (`desiredStatus`, `oldStatus`, `badStatus`) thus
-allowing to query and join different log lines of the `status` field.
-
-Kubernetes objects should be referenced using only their kind, no matter their api group or version. Example argument
-names: `deployment`, `pod`, `node`, `replicaSet`. For objects of unknown type, is ok to log them under `object` key with
-addition of `apiVersion` and `kind` fields describing the k8s object type.
-
-In situations when we want to the log value of the same meaning twice (e.g. transition between state) it is ok to name
-an additional argument based on context, but leaving one most current/correct value with canonical name.
-
-Examples of keys (strongly suggested, will be extended when pattern emerge, no standard schema yet):
-* `err` - error when using `klog.InfoS`. Used for expected errors that are not `klog.ErrorS`.
-* `object` - reference to k8s objects of unknown type. Should be used with `kind` and `apiVersion`.
-* `kind` - kind of k8s object of unknown type.
-* `apiVersion` -  API version of k8s object of unknown type.
-
-Example:
+When deciding on names of arguments you should:
+* Always use [lowerCamelCase], for example use `containerName` and not `container name` or `container_name`.
+* Use [alphanumeric] characters and not should not any special characters like `%$*`, non-latin or unicode characters.
+* Use object kind when referencing Kubernetes objects, for example `deployment`, `pod` and `node`.
+* Describe the type of value stored under they key and avoid to:
+  * Include any of value properties, for example instead of `directory` use `path`.
+  * Describe context how value is used, for example instead of `podIP` use `IP`.
+  * Shorten the name (except acronyms), for example instead of `addr` use `address`
+  * Use ambiguous names, that require falling back on message to undestand what value it is. For example instead of
+    `key` use `cacheKey` or instead of `version` use `dockerVersion`.
```

Comment:
This sort of contradicts the guidance above on "don't describe how the value is used", can we make this clearer?

Reply:
Sorry for not being clear.

Motivation behind this was to avoid overloading some terms that are not really the same thing or cannot be matched or compared. Without this guidance any label key, cache key, annotation key would be put under "key". They are pretty different in what values would you expect and you would never want to compare or match different types.

Similarly to "version", even thou all versions could be using a semantic representation there is no benefit from putting all of them in same key as you would never want to compare  docker version to kubelet version.

I think our overall goal of picking keys is grouping values that we would want to potentially match or compare. Fact that some IP is used by a Pod should not dictate it's name as it would make it hard to correlate all other IP that are not used by Pod. On the other hand "cacheKey" makes sense as you would want to be able to check usage of cache without needing to consider any other type of key.

Please let me know if that make sense as I may be overthinking this.

Reply:
I wrote a suggested update.